### DISCUSSION TITLE
smoother-rebases-with-auto-squashing-git-commits/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#133] [Announcements]
# smoother-rebases-with-auto-squashing-git-commits/

In this post I introduce the --autosquash flag for git interactive rebases, describe what it does and how to format your commits to use it

https://andrewlock.net/smoother-rebases-with-auto-squashing-git-commits/

**Comment by thomaslevesque** [comment]
Thanks! I've been using fixup commits and autosquash for many years, but I wasn't aware of the `fixup -C` option and the `:/message` trick to search for a commit message.

**Comment by ArturDorochowicz** [comment]
Now eagerly awaiting an article on git-absorb
https://github.com/tummychow/git-absorb

**Reply by andrewlock** [reply]
Thanks for this, it's great! 😃 Wrote a post about it here: https://andrewlock.net/super-charging-git-rebase-with-git-absorb/

**Comment by hahn-kev** [comment]
This looks really cool. I just tried this with Rider and didn't have any success. Do I need to restart something to pickup the config change?

**Reply by christian-elsee** [reply]
>Do I need to restart something to pickup the config change?

Yes, restart the workflow by tossing `Rider` and just use the `git` cli, as god intended.

**Comment by christian-elsee** [comment]
TIL! Thanks @andrewlock

**Comment by stephan-koenig** [comment]
Thank you for this! When I run `git rebase -i --autosquash <branch>` the editor opens with no listed commits and just `noop`. If I run `git rebase -i --autosquash HEAD~<number>` it works. Is any of my other git settings interfering?
